Sources/Amplitude/Amplitude.m

```diff
@@ -1669,19 +1669,21 @@ - (NSString *)getAdSupportID {
 }
 
 - (NSString *)getDeviceId {
-    return self.deviceId;
+    return [self initializeDeviceId];
```

Comment:
@crleona So something like this?

Reply:
yeah, this works - we could only call initializeDeviceId on a nil check, but the overhead should be negligible and I doubt we're going to encounter any thread contention.